LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/Nct677X.cs

```diff
@@ -817,11 +817,43 @@ private byte ReadByte(ushort address)
         }
 
         byte page = (byte)(address >> 8);
-        byte index = (byte)(address & 0xFF);
-        Ring0.WriteIoPort(_port + EC_SPACE_PAGE_REGISTER_OFFSET, EC_SPACE_PAGE_SELECT);
+        byte index = (byte)(address & 0xFF);
+
+        //wait for access, access == EC_SPACE_PAGE_SELECT
+        //after 35 retries @ 16ms, abort and force access
+        //after x millseconds timeout abort wait loop
+        byte access;
+        DateTime timeout = DateTime.UtcNow.AddMilliseconds(500);
+        int retries = 35;
+        do
+        {
+            access = Ring0.ReadIoPort(_port + EC_SPACE_PAGE_REGISTER_OFFSET);
+            if (access != EC_SPACE_PAGE_SELECT)
+            {
+                retries--;
+                System.Threading.Thread.Sleep(15);
+                if (DateTime.UtcNow > timeout)
+                {
+                    retries = -1; //timeout elapsed
+                }
+            }
+        }
+        while (access != EC_SPACE_PAGE_SELECT && retries > 0);
+
+        if (retries <= 0)
+        {
+            //timeout: force register access
+            Ring0.WriteIoPort(_port + EC_SPACE_PAGE_REGISTER_OFFSET, EC_SPACE_PAGE_SELECT);
+        }

```

Comment:
This seems unnecessary complicated. This same suggestion applies to WriteByte.

```suggestion
        DateTime timeout = DateTime.UtcNow.AddMilliseconds(500);

        while (true)
        {
            access = Ring0.ReadIoPort(_port + EC_SPACE_PAGE_REGISTER_OFFSET);
            if (access == EC_SPACE_PAGE_SELECT || DateTime.UtcNow > timeout)
                break;                
            
            System.Threading.Thread.Sleep(1);
        }
        
        if (access != EC_SPACE_PAGE_SELECT)
        {
            // Failed to gain access: force register access
            Ring0.WriteIoPort(_port + EC_SPACE_PAGE_REGISTER_OFFSET, EC_SPACE_PAGE_SELECT);
        }
```